### PR TITLE
feat(package): add LoggedInGuard

### PR DESCRIPTION
**Pull request by beeman**
@AnthonyNahas thanks for you comments on #154 

This branch is still WIP but I'd love you to take a look.

I was wondering if you like the name `LoggedInGuard` or if you prefer something else.

It's not tested yet! I tried running the demo locally but did not get it working at first try, will look into that later. Can you shortly share your dev cycle for working on this library?

### CODE REVIEW

File: tslint.json [hunk 1]
```diff
@@ -13,10 +13,6 @@
     "curly": true,
     "eofline": true,
     "forin": true,
-    "import-blacklist": [
-      true,
-      "rxjs"
```
Comment (beeman) [inline tslint.json]: Since RxJS v6 we can remove this.

Comment (AnthonyNahas) [reply tslint.json]: thank you for the info 👍  I will remove that too in other several [packages](https://github.com/angular-material-extensions)

File: src/module/guards/logged-in.guard.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import {
+  AuthProcessService,
+  NgxAuthFirebaseUIConfig,
+  NgxAuthFirebaseUIConfigToken,
+} from '../ngx-auth-firebase-u-i.module'
+
+@Injectable({
+  providedIn: 'root',
+})
+export class LoggedInGuard implements CanActivate {
```
Comment (AnthonyNahas) [inline src/module/guards/logged-in.guard.ts]: What do you think about the name of the guard ... `AuthGuard` vs `LoggedInGuard` ?

File: src/module/guards/logged-in.guard.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  constructor(
+    @Inject(NgxAuthFirebaseUIConfigToken)
+    private config: NgxAuthFirebaseUIConfig,
+    private router: Router,
+    private auth: AuthProcessService,
+  ) {}
+
+  canActivate(): Observable<boolean> {
+    return this.auth.afa.user.pipe(
```
Comment (AnthonyNahas) [inline src/module/guards/logged-in.guard.ts]: Do we need the user `Observable ` or the `authState` `Observable` ? 

I have in my project the following implementation of the `authGuard` 


```typescript
return new Promise((resolve, reject) => {
            this._angularFireAuth.authState.subscribe((user) => {
                if (user) {
                    user.getIdToken()
                        .then((token) => {
                            if (this.checkAccessToken(token)) { // the token can be expired ...
                                resolve(true);
                            }
                            else {
                                resolve(false);
                                // reject("access token invalid");
                            }
                        })
                   // ... other code
                }
```


```ts
private checkAccessToken(accessToken): boolean {
        let jwtHelper: JwtHelper = new JwtHelper();
        jwtHelper.decodeToken(accessToken); //only for debug
        jwtHelper.getTokenExpirationDate(accessToken); //Only for debug --> todo show exp date in the webapp
        return !jwtHelper.isTokenExpired(accessToken);
    }
``
`

which one do you think is more accurate ?

File: src/module/ngx-auth-firebase-u-i.module.ts [hunk 4]
```diff
@@ -41,6 +41,8 @@ export {LegalityDialogComponent} from './components/legality-dialog/legality-dia
 export {AuthProcessService, AuthProvider} from './services/auth-process.service';
 export {FirestoreSyncService} from './services/firestore-sync.service';
 
+// guards
+export {LoggedInGuard} from './guards/logged-in.guard';
```
Comment (AnthonyNahas) [inline src/module/ngx-auth-firebase-u-i.module.ts]: awesome! the guard can now be imported from `ngx-auth-firebaseui`'s module!

However, we need to provide it in `NgxAuthFirebaseUIModule`

similar like below...

```ts
export class NgxAuthFirebaseUIModule {
  static forRoot(configFactory: FirebaseAppConfig,
                 appNameFactory?: () => string,
                 config: NgxAuthFirebaseUIConfig = defaultAuthFirebaseUIConfig): ModuleWithProviders {
    return {
      ngModule: NgxAuthFirebaseUIModule,
      providers:
        [
          {
            provide: FirebaseOptionsToken,
            useValue: configFactory
          },
          {
            provide: FirebaseNameOrConfigToken,
            useFactory: appNameFactory
          },
          {
            provide: NgxAuthFirebaseUIConfigToken,
            useValue: config
          },
          AuthProcessService,
          FirestoreSyncService,
          LoggedInGuard // the guard is now within the maim module provided and can be used
        ],
    };
  }
}
```

File: src/module/ngx-auth-firebase-u-i.module.ts [hunk 5]
```diff
@@ -41,6 +41,8 @@ export {LegalityDialogComponent} from './components/legality-dialog/legality-dia
...
 // interfaces
```
Comment (AnthonyNahas) [inline src/module/ngx-auth-firebase-u-i.module.ts]: @beeman great work 👍 thank you

File: src/module/guards/logged-in.guard.ts (hunk 3 above)
Comment (AnthonyNahas) [reply src/module/guards/logged-in.guard.ts]: the `AngularFireAuth.use` is actually the recommanded way to track and monitor the application's authentication state --> [info](https://github.com/angular/angularfire2/blob/master/docs/auth/getting-started.md) 👍

**Pull request merged**